cmake/ImportLLVM.cmake

```diff
@@ -39,6 +41,12 @@ Once the modules are included we verify the LLVM version is supported and set
 some additional compile definitions.
 #]=======================================================================]
 
+# If we already have the LLVM CMake variables set then we don't need to include from
+# CA_LLVM_INSTALL_DIR
+if (DEFINED LLVM_VERSION_MAJOR)
```

Comment:
Clang also supports building either inside LLVM or outside, it checks it by `CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR`, would it work for us to do it the same way?

Reply:
no, because we promote doing this for building OCK customer targets, so we can't really make this check in the same way. Also, since it's not really an LLVM project as such it feels a bit general.

Reply:
I don't quite get what you mean here, sorry. I know we support at least two ways of building OCK, one is from OCK itself, one is as part of DPC++ with the `native_cpu` plugin, but it sounds like those are not the only two ways?

Reply:
I think if it's already set anyway, we are likely to get into a messy cmake situation if we try to import it here, but I'm open to suggestions.

Reply:
Thinking about this, is there some target that in-LLVM-tree builds will definitely have available, and out-of-LLVM-tree builds can be reasonably assumed won't even if they have imported LLVM? For instance, `if(TARGET llvm-tblgen)` except I think that one might actually also be available as an imported target. Trying to find something better...

Reply:
The only reason I went with this is that we use LLVM_VERSION_MAJOR elsewhere. We do `if(TARGET LLVMCore)` - maybe something like that is guaranteed?